addons/stock/models/stock_scrap.py

```diff
@@ -151,16 +151,12 @@ def action_get_stock_picking(self):
 
     def action_get_stock_move_lines(self):
         action = self.env['ir.actions.act_window']._for_xml_id('stock.stock_move_line_action')
-        action['domain'] = [('move_id', '=', self.move_id.id)]
+        action['domain'] = [('move_id', 'in', self.move_ids.ids)]
         return action
 
-    def action_validate(self):
-        self.ensure_one()
-        if float_is_zero(self.scrap_qty,
-                         precision_rounding=self.product_uom_id.rounding):
-            raise UserError(_('You can only enter positive quantities.'))
+    def check_available_qty(self):
         if self.product_id.type != 'product':
-            return self.do_scrap()
+            return True
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         available_qty = sum(self.env['stock.quant']._gather(self.product_id,
```

Comment:
Maybe you could try with using the qty_available product product. It would avoid the override in mrp